cmd/serve/restic/restic.go

```diff
@@ -341,7 +341,11 @@ func (s *server) serveObject(w http.ResponseWriter, r *http.Request) {
 	o, err := s.newObject(r.Context(), remote)
 	if err != nil {
 		fs.Debugf(remote, "%s request error: %v", r.Method, err)
-		http.Error(w, http.StatusText(http.StatusNotFound), http.StatusNotFound)
+		if err == fs.ErrorObjectNotFound {
```

Comment:
This should probably use `errors.Is` as we do elsewhere in this code.

Reply:
I've also added a commit that replaces the err comparison used when deleting a file.